flutter-idea/src/io/flutter/analytics/FlutterAnalysisServerListener.java

```diff
@@ -342,51 +367,85 @@ public void serverStatus(AnalysisStatus analysisStatus, PubStatus pubStatus) {
+    if (observeThrottling && !IS_TESTING) {
+      long currentTimestamp = System.currentTimeMillis();
+      // Throttle to one report per interval.
+      if (currentTimestamp - generalTimestamp < GENERAL_REPORT_INTERVAL) {
+        return;
+      }
+      generalTimestamp = currentTimestamp;
+    }
+    func.accept(FlutterInitializer.getAnalytics());
+  }
+
+  private void maybeReportErrorCounts() {
+    long currentTimestamp = System.currentTimeMillis();
+    // Send accumulated error counts once every defined interval, plus when the project is closed.
+    if (errorsTimestamp == 0L || currentTimestamp - errorsTimestamp > ERROR_REPORT_INTERVAL || IS_TESTING) {
+      errorsTimestamp = currentTimestamp;
+      Analytics analytics = FlutterInitializer.getAnalytics();
+      LOG.debug(DAS_STATUS_EVENT_TYPE + " " + errorCount + " " + warningCount + " " + hintCount + " " + lintCount);
+      if (errorCount > 0) {
+        analytics.sendEventMetric(DAS_STATUS_EVENT_TYPE, ERRORS, errorCount); // test: serverStatus()
+      }
+      if (warningCount > 0) {
+        analytics.sendEventMetric(DAS_STATUS_EVENT_TYPE, WARNINGS, warningCount); // test: serverStatus()
+      }
+      if (hintCount > 0) {
+        analytics.sendEventMetric(DAS_STATUS_EVENT_TYPE, HINTS, hintCount); // test: serverStatus()
+      }
+      if (lintCount > 0) {
+        analytics.sendEventMetric(DAS_STATUS_EVENT_TYPE, LINTS, lintCount); // test: serverStatus()
+      }
+      errorCount = warningCount = hintCount = lintCount = 0;
+    }
+  }
   private static int extractCount(@NotNull Map<String, Integer> errorCounts, String name) {
     //noinspection Java8MapApi,ConstantConditions
-    return errorCounts.containsKey(AnalysisErrorType.CHECKED_MODE_COMPILE_TIME_ERROR) ? errorCounts.get(
-      AnalysisErrorType.CHECKED_MODE_COMPILE_TIME_ERROR) : 0;
+    return errorCounts.containsKey(name) ? errorCounts.get(name) : 0;
   }
 
   @Override
   public void computedExistingImports(String file, Map<String, Map<String, Set<String>>> existingImports) {
   }
 
-  private static void logCompletion(@NotNull String selection, int prefixLength, @NotNull String eventType) {
-    FlutterInitializer.getAnalytics().sendEventMetric(eventType, selection, prefixLength); // test: acceptedCompletion(), lookupCanceled()
+  private void logCompletion(@NotNull String selection, int prefixLength, @NotNull String eventType) {
+    maybeReport(true, (analytics) -> {
+      LOG.debug(eventType + " " + selection + " " + prefixLength);
+      analytics.sendEventMetric(eventType, selection, prefixLength); // test: acceptedCompletion(), lookupCanceled()
+    });
   }
 
   void logE2ECompletionSuccessMS(long e2eCompletionMS) {
-    FlutterInitializer.getAnalytics().sendTiming(E2E_IJ_COMPLETION_TIME, SUCCESS, e2eCompletionMS); // test: logE2ECompletionSuccessMS()
+    maybeReport(true, (analytics) -> {
```

Comment:
For the completion time, rather than throttling, what you could alternately report a single metric on intellij close that reports the P50, P90, and P95 times for the entire session.
That would make the completion time numbers less noisy than filtering to only report 1 completion per minute.

Reply:
See the comment below for why we cannot do it when exiting.